chain/chain/src/chain.rs

```diff
@@ -2413,6 +2423,7 @@ impl Chain {
                 let shard_uid = self.epoch_manager.shard_id_to_uid(shard_id, epoch_id)?;
                 let flat_storage_manager = self.runtime_adapter.get_flat_storage_manager();
                 flat_storage_manager.update_flat_storage_for_shard(shard_uid, &block)?;
+                self.garbage_collect_memtrie_roots(&block, shard_uid);
```

Comment:
nit: if these are set of instructions we have to execute upon certain event (as it's repeated on line 3601 as well), does it make sense to have them in a separate function?

Reply:
Yeah probably... but it's gonna be a pretty small function; not too sure if that helps with readability. But @Longarithm might have an opinion here - I don't have the context around flat storage.

Reply:
Good question. I also prefer to keep them separate because it looks a bit more readable.